nifi-nar-bundles/nifi-update-attribute-bundle/nifi-update-attribute-processor/src/main/java/org/apache/nifi/processors/attributes/UpdateAttribute.java

```diff
@@ -259,6 +270,14 @@ public void onScheduled(final ProcessContext context) throws IOException {
 
             context.getStateManager().setState(tempMap, Scope.LOCAL);
         }
+
+        defaultActions = getDefaultActions(context.getProperties());
+        debugEnabled = getLogger().isDebugEnabled();
+    }
+
+    @OnUnscheduled
+    public void onUnScheduled() {
+        defaultActions = null;
```

Comment:
This needs to be in the "OnStopped", not the "OnUnscheduled". "OnUnscheduled" will get executed once the user hits the "stop" button and there could still be threads running, leading to the NPE below. ("OnStopped" runs once all threads running the ontrigger have stopped)

![screen shot 2016-12-28 at 11 28 11 am](https://cloud.githubusercontent.com/assets/11302527/21526179/39ea5e8c-ccf0-11e6-8653-a3d8e7e02f04.png)